packages/cli/src/evaluation/test-definitions.controller.ee.ts

```diff
@@ -0,0 +1,87 @@
+import { Get, Post, Patch, RestController, Delete } from '@/decorators';
+import { BadRequestError } from '@/errors/response-errors/bad-request.error';
+import { NotFoundError } from '@/errors/response-errors/not-found.error';
+import { listQueryMiddleware } from '@/middlewares';
+import { getSharedWorkflowIds } from '@/public-api/v1/handlers/workflows/workflows.service';
+import { isPositiveInteger } from '@/utils';
+
+import { TestDefinitionsService } from './test-definitions.service.ee';
+import { TestDefinitionsRequest } from './test-definitions.types.ee';
+
+@RestController('/evaluation/test-definitions')
+export class TestDefinitionsController {
+	constructor(private readonly testsService: TestDefinitionsService) {}
+
+	@Get('/', { middlewares: listQueryMiddleware })
+	async getMany(req: TestDefinitionsRequest.GetMany) {
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		return await this.testsService.getMany(req.listQueryOptions, workflowIds);
+	}
+
+	@Get('/:id')
+	async getOne(req: TestDefinitionsRequest.GetOne) {
+		if (!isPositiveInteger(req.params.id)) {
+			throw new BadRequestError('Test ID is not a number');
+		}
+
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		const testDefinition = await this.testsService.findOne(Number(req.params.id), workflowIds);
+
+		if (!testDefinition) throw new NotFoundError('Test not found');
+
+		return testDefinition;
+	}
+
+	@Post('/')
+	async create(req: TestDefinitionsRequest.Create) {
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		if (!workflowIds.includes(req.body.workflowId)) {
+			throw new BadRequestError('User does not have access to the workflow');
+		}
+
+		if (req.body.evaluationWorkflowId && !workflowIds.includes(req.body.evaluationWorkflowId)) {
+			throw new BadRequestError('User does not have access to the evaluation workflow');
+		}
+
+		return await this.testsService.save(this.testsService.toEntity(req.body));
+	}
+
+	@Delete('/:id')
+	async delete(req: TestDefinitionsRequest.Delete) {
+		if (!isPositiveInteger(req.params.id)) {
+			throw new BadRequestError('Test ID is not a number');
+		}
+
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		if (workflowIds.length === 0) throw new NotFoundError('Test not found');
```

Comment:
Why is the test not found if there aren't any workflows found?

Reply:
It means the user has no access to any workflow. This means user can not have access to any test either (because access to test is decided based on access to the workflow under test, and user do not have any). Usually it's a good practice to return "not found" errors instead of "access denied" to avoid exposing ids of the entities user has no access to (enumeration attack). But this is discussable, of course. What do you think?

Reply:
In that case, I'd probably rephrase the error-message, it might be confusing if it would pop up somewhere